lib/Base/Organization/Thread.vala

```diff
@@ -27,11 +27,42 @@ using GLib;
  * "main post". It will display all posts preceding the main post until
  * the top one, as well as all replies to the main post.
  */
-public abstract class Backend.Thread : Backend.Collection {
+public abstract class Backend.Thread : Backend.Collection<Post>, Backend.PullableCollection<Post> {
+
+  /**
+   * The session used to pull posts.
+   */
+  public Session session { get; construct; }
 
   /**
    * The post from which the thread is build.
    */
   public Post main_post { get; construct; }
 
+  /**
+   * The id of the newest item in the collection.
+   */
+  protected string? newest_item_id { get; set; default = null; }
+
+  /**
+   * Calls the API to retrieve all items from this Collection.
+   *
+   * @throws Error Any error while accessing the API and pulling the items.
+   */
+  public abstract async void pull_items () throws Error;
+
+  /**
+   * Used to compares two iterators in the list when sorting.
+   *
+   * FIXME Implementation is postponed until the end of this project.
```

Comment:
What's the plan for thread sorting? Because it doesn't appear to be a default that's overridden in the Mastodon classes.

Reply:
Oh, seems like I completely forgot about that...

I think sorting should be quite similar to `ReversePostList`, but chronological, but I have to look into that soon.